### PR TITLE
working copy with alliance manager and user manager

### PR DESCRIPTION
**Pull request by maximenkom**
If leader logins he redirects to page where he can manage users of his clan
If admin logins he can manage alliances

### CODE REVIEW

File: src/main/java/ua/cv/tim/controller/UserController.java [hunk 1]
```diff
@@ -58,20 +55,18 @@ public String showAdminMainPage() {
...
-	@RequestMapping(value = "/add", method = RequestMethod.GET)
-	public ModelAndView addUserForm() {
-
-		ModelAndView model = new ModelAndView("addUser.html");
-		return model;
-	}
 
 	@RequestMapping(value = "/submitUserForm",method = RequestMethod.POST, produces = {MediaType.APPLICATION_JSON_VALUE})
 	public ResponseEntity<UserDTO> submitUserForm(@RequestBody UserDTO userDTO) {
 		logger.info("UserMail {}" ,userDTO.getEmail());
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/controller/UserController.java]: I think better to join in one line

File: src/main/java/ua/cv/tim/controller/UserController.java [hunk 2]
```diff
@@ -132,5 +127,49 @@ public ModelAndView getAllUsers2() {
...
+	public ResponseEntity<UserInfoDTO> addAllianceMember(@RequestBody UserInfoDTO member) {
+		logger.info("UserController.addAllianceMember() method is working. Alliance name: {}", member.toString());
+		User user = new User();
+		user.setLogin(member.getLogin());
+		user.setEmail(member.getEmail());
+		if (!userService.isUnique(user)) {
+			logger.error("UserController.addAllianceMember: {}", "user is not unique");
+			return new ResponseEntity<>(HttpStatus.NOT_ACCEPTABLE);
+		}
+		try {
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/controller/UserController.java]: clean code: try/catch in separate method

File: src/main/java/ua/cv/tim/dao/hibernate/UserDaoImpl.java [hunk 3]
```diff
@@ -43,9 +56,7 @@ public User getWithRolesById(String id) {
     }
 
     public User getByMail(String mail, String uuid) {
-        System.out.println("Dao get by email");
-        System.out.println("uuid = : " + uuid);
-
+        logger.info("UserDao.getByMail  started mail is {} uuid is {} ", mail,uuid);
         Session session = getCurrentSession();
```
Comment (ViacheslavKruhlov) [inline src/main/java/ua/cv/tim/dao/hibernate/UserDaoImpl.java:60]: I think better do not create Session instance and use getCurrentSession() method instead when create Query instance

File: src/main/java/ua/cv/tim/controller/UserController.java (hunk 2 above)
Comment (maximenkom) [reply src/main/java/ua/cv/tim/controller/UserController.java]: fixed

File: src/main/java/ua/cv/tim/controller/UserController.java (hunk 1 above)
Comment (maximenkom) [reply src/main/java/ua/cv/tim/controller/UserController.java]: this was old method, now it deleted already

**Comment by maximenkom** [comment]
Added new ExceptionHandlerClass in controllers folder, now its successfully handle exceptions for addAllianceMember method, fixed mistakes pointed out by collegues

**Pull request merged**